src/runtime/ArrayBufferObject.cpp

```diff
@@ -105,7 +105,7 @@ void ArrayBufferObject::detachArrayBuffer()
 
     if (m_backingStore && !m_mayPointsSharedBackingStore) {
         // if backingstore is definitely not shared, we deallocate the backingstore immediately.
-        m_backingStore.value()->deallocate();
+        delete m_backingStore.value();
```

Comment:
Is it safe to delete `BackingStore` directly without `GCFree`?
`BackingStore` is defined as GC-object, so I'm wondering that delete operation is safe for this.

Reply:
It is safe.
class gc overrides "operator delete"
```
void gc::operator delete(void* obj) { GC_FREE(obj); }
```
the operator function guarantees memory safety